src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx

```diff
@@ -0,0 +1,136 @@
+import type {ForwardedRef} from 'react';
+import React, {forwardRef, useCallback} from 'react';
+import {View} from 'react-native';
+// eslint-disable-next-line no-restricted-imports
+import type {ScrollView as ScrollViewRN} from 'react-native';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import ScrollView from '@components/ScrollView';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type {ApprovalWorkflowOnyx} from '@src/types/onyx';
+import type {Approver} from '@src/types/onyx/ApprovalWorkflow';
+
+type ApprovalWorkflowEditorProps = {
+    approvalWorkflow: ApprovalWorkflowOnyx;
+    policyID: string;
+};
+
+function ApprovalWorkflowEditor({approvalWorkflow, policyID}: ApprovalWorkflowEditorProps, ref: ForwardedRef<ScrollViewRN>) {
+    const styles = useThemeStyles();
+    const {translate, toLocaleOrdinal} = useLocalize();
+
+    const approverDescription = useCallback(
+        (index: number) =>
+            approvalWorkflow.approvers.length > 1 ? `${toLocaleOrdinal(index + 1, true)} ${translate('workflowsPage.approver').toLowerCase()}` : `${translate('workflowsPage.approver')}`,
+        [approvalWorkflow.approvers.length, toLocaleOrdinal, translate],
+    );
+    const members = approvalWorkflow.isDefault ? translate('workspace.common.everyone') : approvalWorkflow.members.map((m) => m.displayName).join(', ');
+
+    const approverErrorMessage = useCallback(
+        (approver: Approver | undefined, approverIndex: number) => {
+            const previousApprover = approvalWorkflow.approvers.slice(0, approverIndex).filter(Boolean).at(-1);
+            const error = approvalWorkflow?.errors?.[`approver-${approverIndex}`];
+
+            if (!error) {
+                return;
+            }
+
+            if (error === 'workflowsPage.approverCircularReference') {
+                if (!previousApprover || !approver) {
+                    return;
+                }
+                return translate('workflowsPage.approverCircularReference', {
+                    name1: approver.displayName,
+                    name2: previousApprover.displayName,
+                });
+            }
+
+            return translate(error);
+        },
+        [approvalWorkflow.approvers, approvalWorkflow.errors, translate],
+    );
+
+    const approverHintMessage = useCallback(
+        (approver: Approver | undefined, approverIndex: number) => {
+            const previousApprover = approvalWorkflow.approvers.slice(0, approverIndex).filter(Boolean).at(-1);
+            if (approver?.isInMultipleWorkflows && approver.email === previousApprover?.forwardsTo) {
```

Comment:
@blazejkustra It fails in this case. We shouldn't display hint text

![Uploading Screenshot 2024-08-14 at 15.34.35.png…]()

Reply:
![image](https://github.com/user-attachments/assets/f17bd42a-ab7b-48ba-ad23-bc27a670722b)

Reply:
It is a valid point @DylanDylann, I think we should change the hint message to something simpler as the current message is making the display logic impossible (there are too many cases which we can't handle with just one hint message).

@JmillsExpensify what do you think to change the hint message to something more generic?
```
dylandylan9112+9366@gmail.com is already used in a separate workflow. If you change this approval relationship, all other workflows will be updated.
```